### PR TITLE
Fast search

### PR DESCRIPTION
**Pull request by y-lohse**
So the search now uses Cozy Client. I've made the following changes to the client:

- Make sure the cozy url has a protocol.
- Since the doctype `io.cozy.files` can't be written *to*, we only want to replicate *from* the cozy instance. I've added methods that allow to do one way syncs.

### CODE REVIEW

File: src/drive/ducks/services/components/SuggestionProvider.jsx [hunk 1]
```diff
@@ -40,23 +41,11 @@ class SuggestionProvider extends React.Component {
   // fetches pretty much all the files and preloads FuzzyPathSearch
```
Comment (enguerran) [inline src/drive/ducks/services/components/SuggestionProvider.jsx]: Shall we keep this comment?

File: src/lib/cozy-client/DataAccessFacade.js [hunk 2]
```diff
@@ -18,7 +22,13 @@ export default class DataAccessFacade {
   }
 
   setup(cozyUrl, options) {
-    const config = { cozyURL: cozyUrl, ...options }
+    // prepend the protocol if there isn't one, required for pouchdb
+    if (/^http(s):/.test(cozyUrl) === false) {
```
Comment (enguerran) [inline src/lib/cozy-client/DataAccessFacade.js]: `test` is already returning a boolean, so I guess we should write
```js
if (!/^http(s):/.test(cozyUrl) {
```

Comment (enguerran) [reply src/lib/cozy-client/DataAccessFacade.js]: What is the difference between `/^http(s):/` and `/^https:/`?

File: src/lib/cozy-client/DataAccessFacade.js [hunk 3]
```diff
@@ -18,7 +22,13 @@ export default class DataAccessFacade {
   }
 
   setup(cozyUrl, options) {
```
Comment (enguerran) [inline src/lib/cozy-client/DataAccessFacade.js]: Shall we change the call of `setup(cozyUrl, options)` to be sure the `cozyUrl` contains a protocol?

File: src/lib/cozy-client/DataAccessFacade.js (hunk 2 above)
Comment (y-lohse) [reply src/lib/cozy-client/DataAccessFacade.js]: You're right about the `false` comparison :). `/^http(s):/` allows for both `http` and `https` protocol.

File: src/drive/ducks/services/components/SuggestionProvider.jsx (hunk 1 above)
Comment (y-lohse) [reply src/drive/ducks/services/components/SuggestionProvider.jsx]: I would say yes? I feel like it's a good one line summary of the next 30 lines of code.

**Pull request merged**